### PR TITLE
(FM-8079) Resource API and Transports Hands-on-Lab

### PR DESCRIPTION
**Pull request by DavidS**

### CODE REVIEW

**Comment by DavidS** [comment]
current build failure presumed fixed with rebasing on top of #184

File: docs/hands-on-lab/01-installing-prereqs.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Install Puppet Development Kit (PDK) and other tools
+
+To start, install Puppet Development Kit (PDK), which provides all the necessary tools and libraries to build and test modules. We also recommend an emulator for the target device, a code editor with good Ruby and Puppet support, and git — a version control system to keep track of your progress.
+
+1. [Download PDK](https://puppet.com/download-puppet-development-kit) on your platform of choice.
+
+2. If you do not have a Philips HUE hub available, you can download the [Hue-Emulator](https://github.com/SteveyO/Hue-Emulator/raw/master/HueEmulator-v0.8.jar). You need to have Java installed to run this.
```
Comment (clairecadman) [inline docs/hands-on-lab/01-installing-prereqs.md]: Is hub the right word here? Or is it meant to be bulb? If it's correct please ignore!

Comment (clairecadman) [reply docs/hands-on-lab/01-installing-prereqs.md]: Assuming it's fine as it's used throughout.

File: docs/hands-on-lab/02-connecting-to-the-lightbulbs.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Connecting to the Lightbulbs
+
+While there are no technical restrictions on the kinds of remote devices or APIs you can connect to with transports, for the purpose of this workshop we are connecting to a Philips HUE hub and make some colourful wireless lightbulbs light up. If you (understandably) do not have physical devices available, you can get going with the Hue Emulator.
+
+## Hue Emulator
+
+Download the [HueEmulator-v0.8.jar](https://github.com/SteveyO/Hue-Emulator/blob/master/HueEmulator-v0.8.jar) from [SteveyO/Hue-Emulator](https://github.com/SteveyO/Hue-Emulator).
```
Comment (clairecadman) [inline docs/hands-on-lab/02-connecting-to-the-lightbulbs.md]: Won't they have already done this in the installing section?

File: docs/hands-on-lab/01-installing-prereqs.md (hunk 1 above)
Comment (DavidS) [reply docs/hands-on-lab/01-installing-prereqs.md]: In truth, you'd need both the hub and the bulbs.

File: docs/hands-on-lab/02-connecting-to-the-lightbulbs.md (hunk 2 above)
Comment (DavidS) [reply docs/hands-on-lab/02-connecting-to-the-lightbulbs.md]: that's a duplicate, yeah.

File: docs/hands-on-lab/05-implementing-the-transport-hints.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+## Implementing the transport - Exercise
```
Comment (clairecadman) [inline docs/hands-on-lab/05-implementing-the-transport-hints.md]: @DavidS I'm confused, what is the difference between this section and 'Implementing the transport' below? The other one seems like it is the exercise.

**Comment by clairecadman** [comment]
@DavidS I've finished an initial copy edit of this, but I'll take another look when it's complete. When you do plane to finish the TODO parts? Thanks!

File: docs/hands-on-lab/01-installing-prereqs.md (hunk 1 above)
Comment (clairecadman) [reply docs/hands-on-lab/01-installing-prereqs.md]: Do we need to say that? Or is it fine in this instance to just say hub?

File: docs/hands-on-lab/04-adding-a-new-transport.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Add a new transport
 
-[Eventually](https://github.com/puppetlabs/pdk/pull/666) there will be a `pdk new transport`, for now you'll need to copy in a few files for this workshop.
+[Eventually](https://github.com/puppetlabs/pdk/pull/666) there will be a `pdk new transport` command. For now, you'll need to copy the files below.
```
Comment (DavidS) [inline docs/hands-on-lab/04-adding-a-new-transport.md]: This has been released with pdk 1.12.0

File: docs/hands-on-lab/05-implementing-the-transport-hints.md (hunk 3 above)
Comment (DavidS) [reply docs/hands-on-lab/05-implementing-the-transport-hints.md]: These are additional exercise steps that can be used to deepen the reader's understanding and comfort with the code. They're not necessary for completing the Lab.

File: docs/hands-on-lab/01-installing-prereqs.md (hunk 1 above)
Comment (DavidS) [reply docs/hands-on-lab/01-installing-prereqs.md]: I'll also add "and bulbs" for clarity.

**Comment by DavidS** [comment]
Rebased and finalized.

**Pull request merged**